### PR TITLE
[FLOC-2910] Factor out code for running a registry and pushing an image

### PR DESCRIPTION
**Pull request by exarkun**
Fixes https://clusterhq.atlassian.net/browse/FLOC-2910

This will be used to help resolve FLOC-2911 and FLOC-2912, at least.

Note diff is wrong, it includes already-merged changes from #1871

### CODE REVIEW

File: flocker/node/functional/test_docker.py [hunk 1]
```diff
@@ -503,6 +493,77 @@ def test_pull_image_if_necessary(self):
         d.addCallback(lambda _: self.assertTrue(docker.inspect_image(image)))
         return d
 
+    def push_to_registry(self, image_name, registry):
+        """
+        Push an image identify by a local tag to the given registry.
```
Comment (adamtheturtle) [inline flocker/node/functional/test_docker.py]: identify > identified

File: flocker/node/functional/test_docker.py [hunk 2]
```diff
@@ -507,6 +493,77 @@ def test_pull_image_if_necessary(self):
...
+
+        :param unicode image_name: The local tag which identifies the image to
+            push.
+        :param Registry registry: The registry to which to push the image.
+
+        :return: A ``DockerImage`` describing the image in the registry.  Note
+            in particular the tag of the image in the registry will differ from
+            the local tag of the image.
+        """
+        registry_name = random_name(self).lower()
```
Comment (itamarst) [inline flocker/node/functional/test_docker.py]: Maybe have the pull in here, just to make sure it always works?

**Comment by itamarst** [comment]
Looks fine, if you can get buildbot to be green...

File: flocker/node/functional/test_docker.py (hunk 2 above)
Comment (exarkun) [inline flocker/node/functional/test_docker.py]: I'm not sure what you mean.

**Pull request merged**